apis/tempo/v1alpha1/tempomonolithic_webhook.go

```diff
@@ -0,0 +1,77 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks.
+func (r *TempoMonolithic) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type.
+func (r *TempoMonolithic) Default() {
+	log := ctrl.Log.WithName("tempomonolithic-webhook")
```

Comment:
how is this rendered in the logs? Isn't it too long?

Reply:
It'll print this line:
```
{"level":"debug","ts":"2024-01-15T19:07:08.746058013+01:00","logger":"tempomonolithic-webhook","msg":"running defaulter webhook","name":"sample"}
```
if debug logs are enabled (`go run ./main.go --zap-log-level=debug start`).

Reply:
But as we'll set the defaults in the reconcile loop now, I'm removing this log statement.

I'll keep it there for the validating webhook, which is still in use.